RELEASE/scripts/autoscend/auto_quest_level_12.ash

```diff
@@ -1290,6 +1773,169 @@ boolean L12_themtharHills()
 	return true;
 }
 
+boolean LX_chaosButterfly()
+{
+	if(auto_my_path() == "Bees Hate You" || auto_my_path() == "Pocket Familiars")
+	{
+		return false;
+	}
+	if(get_property("chaosButterflyThrown").to_boolean())
+	{
+		return false;
+	}
+	if(!is_unrestricted($item[chaos butterfly]))
+	{
+		return false;
+	}
+	if(internalQuestStatus("questL10Garbage") < 8)
+	{
+		return false;
+	}
+	
+	//softcore pull
+	if(canPull($item[chaos butterfly]) && !get_property("chaosButterflyThrown").to_boolean() && item_amount($item[chaos butterfly]) == 0)
+	{
+		if(pullXWhenHaveY($item[chaos butterfly], 1, 0))
+		{
+			return true;
+		}
+		else
+		{
+			auto_log_warning("failed to pull [chaos butterfly] for some reason", "red");
+		}
+	}
+	
+	//fight possibility giant for chaos butterfly if profitable
+	if(!get_property("chaosButterflyThrown").to_boolean() && item_amount($item[chaos butterfly]) == 0 && auto_warAdvReqCB() < 15)
+	{
+		if(autoAdv(1, $location[The Castle in the Clouds in the Sky (Ground Floor)]))
+		{
+			return true;
+		}
+		else
+		{
+			auto_log_warning("For some reason failed to adventure in [The Castle in the Clouds in the Sky (Ground Floor)] for a [chaos butterfly]... skipping", "red");
+		}
+	}
+	
+	//if you just now acquired [chaos butterfly] then it should be used in barn to save 1 adv.
+	if(!get_property("chaosButterflyThrown").to_boolean() && item_amount($item[chaos butterfly]) > 0 && auto_my_path() != "Pocket Familiars")
+	{
+		if($location[McMillicancuddy's Barn].turns_spent > 0)
+		{
+			auto_log_warning("I seem to have spent turns in [McMillicancuddy's Barn] without using the [chaos butterfly] which I have. Something is not right...", "red");
+		}
+		else
+		{
+			auto_log_warning("Looks like I should use the [chaos butterfly] in [McMillicancuddy's Barn]", "blue");
+			return autoAdv(1, $location[McMillicancuddy's Barn]);
+		}
+	}
```

Comment:
another nitpick: technically you can use the chaos buttefly in any zone before doing the farm but it probably makes sense to use it there on the first duck you encounter so we're not having to do weird tracking stuff earlier.

Reply:
To clarify, the combat code does indeed use the chaos butterfly in any zone.

This bit of code is for when you just pulled a chaos butterfly right now and this is your last quest (currently the war is done last). by using it on the barn instead of against a hippy in the battlefield you save 1 adv.

Reply:
Ah nice, that's pretty smart. Ignore my comment then.